### PR TITLE
Add resource format

### PR DESCRIPTION
**Pull request by degregat**

### CODE REVIEW

File: resource.bnf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+finite-field-element ::= <u256>
+
+;; logic and value commitments are unique (up to hash collision) identifiers for program field elements
+logic ::= <program-field-element>
+
+value ::= <program-field-element>
+
+;; element of a variable size field
+;; 4 bytes + n bytes
+program-field-element ::= <length> <byte>*
```
Comment (mariari) [inline resource.bnf:31]: I don't see these used  in the resource data type at all, maybe belong elsewhere?

Comment (degregat) [reply resource.bnf:31]: These would be the plaintext representations of the field elements, which are the things the "VM" would need to load into memory if it were to operate on the resources, e.g. computing proofs and such.

Another way to see it is: The finite-field-elements are used as keys in a lookup table to retrieve the program-field-elements. 

If you think we should move that to another file, we can do that, to me it just seemed most usefully grouped with the resource used to communicate their keys.

Comment (mariari) [reply resource.bnf:31]: Might be good to write it in a readme or give a sister file, that sort of explains why better.

Thanks for writing up the BNF

Comment (degregat) [reply resource.bnf:31]: We can move these to data-storage.bnf

**Pull request merged**